src/components/ui/ConditionalButton.tsx

```diff
@@ -0,0 +1,111 @@
+/**
+ * This program has been developed by students from the bachelor Computer Science at Utrecht University within the Software Project course.
+ * © Copyright Utrecht University (Department of Information and Computing Sciences)
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import React from 'react';
+import { Button, ButtonProps } from '@/src/components/ui/Button';
+import { useWeb3Modal } from '@web3modal/react';
+import { HiOutlineExclamationCircle } from 'react-icons/hi2';
+
+export type Condition = {
+  enabled: boolean;
+  content: JSX.Element;
+};
+
+export interface ConditionalButtonProps extends ButtonProps {
+  conditions: Condition[];
+}
+
+export const ConditionalButton = React.forwardRef<
+  HTMLButtonElement,
+  ConditionalButtonProps
+>(
+  (
+    {
+      conditions,
+      className,
+      icon,
+      iconNode,
+      variant,
+      size,
+      label,
+      children,
```

Comment:
Is there a reason these props are not gathered into `...props` and passed to the Button like that?

Reply:
Nope, that's a better solution for sure